lib/model/unittest/CModelTestFixtureBase.h

```diff
@@ -220,8 +222,48 @@ class CModelTestFixtureBase {
                                       const std::string& overFieldName,
                                       TKeyCompareFunc keyCompare);
 
+    template<typename T>
+    void makeModelT(const ml::model::SModelParams& params,
+                    const ml::model_t::TFeatureVec& features,
+                    ml::core_t::TTime startTime,
+                    ml::model_t::EModelType modelType,
+                    ml::model::CModelFactory::TDataGathererPtr& gatherer,
+                    ml::model::CModelFactory::TModelPtr& model,
+                    TOptionalUInt sampleCount = TOptionalUInt(),
+                    const std::string& summaryCountField = EMPTY_STRING) {
+        if (m_InterimBucketCorrector == nullptr) {
+            m_InterimBucketCorrector =
+                std::make_shared<ml::model::CInterimBucketCorrector>(params.s_BucketLength);
+        }
+        if (m_Factory == nullptr) {
+            m_Factory.reset(new T(params, m_InterimBucketCorrector,
+                                  summaryCountField.empty() ? ml::model_t::E_None
+                                                            : ml::model_t::E_Manual,
+                                  summaryCountField));
+            m_Factory->features(features);
+        }
+        ml::model::CModelFactory::SGathererInitializationData initData(startTime);
+        if (sampleCount) {
+            initData.s_SampleOverrideCount = *sampleCount;
+        }
+        gatherer.reset(m_Factory->makeDataGatherer(initData));
+        model.reset(m_Factory->makeModel({gatherer}));
+
+        BOOST_TEST_REQUIRE(model);
+        BOOST_REQUIRE_EQUAL(modelType, model->category());
+        BOOST_REQUIRE_EQUAL(params.s_BucketLength, model->bucketLength());
+    }
+
+protected:
+    using TInterimBucketCorrectorPtr = std::shared_ptr<ml::model::CInterimBucketCorrector>;
+    using TModelFactoryPtr = boost::shared_ptr<ml::model::CModelFactory>;
```

Comment:
Why is this using `boost::shared_ptr`?  If there's a good reason, e.g. it's used with some other Boost library that has `boost::shared_ptr` in its interface, please add a comment.